### ISSUE TITLE
Python bazel distrib tests failing with protobuf 22.x

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#32778]
This is blocks our upgrade to protobuf 22.x

On my development branch https://github.com/jtattermusch/grpc/tree/python_distribtest_import_repro (where most other test are green), when I run the python bazel distribtest, I get following failures for `//:import_test` and `//:import_moved_test` (other bazel distribtests are passing).

```
[3,804 / 3,806] 3 / 5 tests; [Prepa] Testing //:import_test
FAIL: //:import_test (see /root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/execroot/python_test_repo/bazel-out/k8-fastbuild/testlogs/import_test/test.log)
INFO: From Testing //:import_test:
==================== Test output for //:import_test:
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/sandbox/processwrapper-sandbox/2407/execroot/python_test_repo/bazel-out/k8-fastbuild/bin/import_test.runfiles/python_test_repo/helloworld.py", line 22, in <module>
    from google.protobuf import duration_pb2
  File "/root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/sandbox/processwrapper-sandbox/2407/execroot/python_test_repo/bazel-out/k8-fastbuild/bin/import_test.runfiles/python_test_repo/external/com_google_protobuf/src/google/protobuf/_virtual_imports/duration_proto/google/protobuf/duration_pb2.py", line 5, in <module>
    from google.protobuf.internal import builder as _builder
ModuleNotFoundError: No module named 'google.protobuf.internal'
================================================================================
FAIL: //:import_moved_test (see /root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/execroot/python_test_repo/bazel-out/k8-fastbuild/testlogs/import_moved_test/test.log)
INFO: From Testing //:import_moved_test:
==================== Test output for //:import_moved_test:
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/sandbox/processwrapper-sandbox/2408/execroot/python_test_repo/bazel-out/k8-fastbuild/bin/import_moved_test.runfiles/python_test_repo/helloworld_moved.py", line 25, in <module>
    from google.protobuf import duration_pb2
  File "/root/.cache/bazel/_bazel_root/7bb51ef4c4ce2ac55dce6a30e338cda8/sandbox/processwrapper-sandbox/2408/execroot/python_test_repo/bazel-out/k8-fastbuild/bin/import_moved_test.runfiles/python_test_repo/external/com_google_protobuf/src/google/protobuf/_virtual_imports/duration_proto/google/protobuf/duration_pb2.py", line 5, in <module>
    from google.protobuf.internal import builder as _builder
ModuleNotFoundError: No module named 'google.protobuf.internal'
================================================================================
INFO: Elapsed time: 276.522s, Critical Path: 65.09s
INFO: 3805 processes: 1395 internal, 2410 processwrapper-sandbox.
INFO: Build completed, 2 tests FAILED, 3805 total actions
```
https://source.cloud.google.com/results/invocations/f177d683-7430-442e-bb06-f1417f6baadb/log

**Comment by jtattermusch** [comment]
The issue might be somewhat related to https://github.com/protocolbuffers/protobuf/issues/12314 (but I'm not sure at this point)

**Comment by gnossen** [comment]
I started investigating this issue and will continue tomorrow. It looks like the issue is that the `sys.path` /  `PYTHONPATH` order was changed somehow. The module path for `google.protobuf` ends up being the directory associated with the generated `duration_pb2` module rather than the protobuf Python `py_library` target as it should be. The proper `google.protobuf` `__init__.py` file declares itself a namespace package so that multiple packages can share the `google.protobuf` module, but since `duration_pb2` is now first in the list and it does not declare `google.protobuf` to be a namespace package, the resolution stops there and the Python protobuf module's directory is not included in the search for `google.protobuf.internal`.

I don't have any immediate ideas about what might have caused this, but I will comb through the diff tomorrow.
